### DISCUSSION TITLE
How to create dataset to train table recognition model?

### DISCUSSION THREAD

**Discussion by johntoma** [PaddlePaddle/PaddleOCR#8672] [Q&A]
Hello! I intend to create a new dataset using my own set of table images to train one of the table recognition models, but I don't know where to start or what tools I should be using. I also need to include empty cells in the labelling as they are very important for my use case. I know there is a tool "PPOCRLabelv2" that can do some kind of automatic labelling, but when I try to use the "Table Recognition" feature, it doesn't seem to label the cells, but rather the lines of text for some reason.

Can anyone give me some guidance on where I should start, or maybe some examples I can take a look at?

Thanks!

**Comment by TasneemVKhan** [comment]
Hi, even we intend to create our set of table images to train the ppstructure model. 

I want to label my own dataset and train the table recognition model but haven't found a proper path for it. 

I was able to label a few of my table images but the application gives an error and shuts itself down.  Haven't been successful. 

Use this [link](https://github.com/PaddlePaddle/PaddleOCR/tree/release/2.6/PPOCRLabel) to help you succeed in your goal. If any errors you can ask for help

**Reply by johntoma** [reply]
Yeah I am aware of that tool. Unfortunately I'm also experiencing some issues with the tool crashing. I've submitted issues related to that but haven't received replies from the PaddleOCR team yet.

**Reply by TasneemVKhan** [reply]
Yeah I have faced the same issue. Refer to this [link]  (https://github.com/PaddlePaddle/PaddleOCR/discussions/8015#discussioncomment-4664037). Ongoing discussion regarding the same issue of application crashing

**Comment by AkshatHotCode** [comment]
Did you get any reply from them, I am too trying to fine-tune the model with my own set of data, but am unable to process it. Can anyone of you help me with the same?

**Reply by TasneemVKhan** [reply]
What are the errors which you are currently facing ?

**Reply by AkshatHotCode** [reply]
I haven't started yet. I have some images containing tables in it, and I want to fine-tune the model based on that dataset.
For that, I have to first start with annotating the images, is there any tool you guys are using to annotate the coordinates of the tables in the images, or anything?

**Reply by TasneemVKhan** [reply]
PPOCRLabel is the annotations tool used to generate the dataset for table recognition. 

PPOCRLabel has many bugs, even I haven't been successful in it

**Reply by AkshatHotCode** [reply]
I am able to annotate the dataset on my machine, and i am getting the final output too as Lable.txt.
Now can you help me how can we use this annotations to fine tune the model? https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/table_recognition_en.md In this link, the format of dataset is different of what we are getting from PPOCRLabel.

**Reply by TasneemVKhan** [reply]
Are you able to generate a gt.txt file ? 

Use train.py and whichever config file you want to use make changes in it and run it through the cmd

**Reply by AkshatHotCode** [reply]
I was able to get Label.txt.
![image](https://user-images.githubusercontent.com/95849362/212903563-8602ee30-fe87-4c6f-8648-8abc8021e4ee.png)

**Reply by SivaKanishka** [reply]
@AkshatHotCode, were you able to identify a way to use Lable.txt for fine tuning the table detection model. I feel there are some missing pieces in the documentation.

**Comment by johntoma** [comment]
@AkshatHotCode @TasneemVKhan Hello, when you annotated your dataset, did you also label the empty cells as well? It's still unclear to me whether or not we must label the empty cells for the training data. I've submitted two tickets about this but unfortunately the paddle team arent responding to them.

**Comment by Shreejan-git** [comment]
Annotating the whole table (including the blank part as well) and letting the paddleocrlabel to generate gt file gives completely messed HTML. I do not know what should we exactly do!
